src/types/index.ts

```diff
@@ -26,12 +26,11 @@ export interface WalletConnection {
   l2Signer: L2Signer;
 }
 
-export type EthNetwork = 'dev' | 'ropsten' | 'mainnet';
-
 export interface Config {
   api: Configuration;
```

Comment:
`Config` interface with an `api` property of type `Configuration` seems a little suspect

Reply:
Type Configuration is from the open api gen code so we wouldn't change it.
Contract Addresses and ChainID are not related to api config but still part of config we want to cache for the user so they can use those values.

Reply:
@zatlite we don't need to expose the generated API library's `Configuration` type here at all. It's only needed to create generated API objects inside of Workflows's constructor. The `Configuration` object can also be created there and hidden from our Config interface. That way, if we change the underlying codegen tool or template the exposed interface isn't going to be affected.

Reply:
No, I think we want to expose it because we're using this function to tack on the x-sdk-version http header for tracking. The configuration object is needed when users want to connect to the generated api client so user will need to build one even when they're not using workflows and just want to use the api client. 

We could shift the default http headers into openapi gen templates so that it's not dependent on the configuration object but that would mean we take on another maintenance task for keeping the templates working when/if openapi gen code updates with breaking changes.

Reply:
Config interface replaced with ImmutableXConfiguration interface.
[cf33110](https://github.com/immutable/imx-core-sdk/pull/121/commits/cf33110d1dc6503c1e747dfced4fff0ec57bb536)